### PR TITLE
Basic query routing

### PR DESCRIPTION
**Pull request by vogti**
This PR implements a basic approach for routing queries.

### CODE REVIEW

File: core/src/main/java/org/polypheny/db/schema/ModifiableTable.java [hunk 1]
```diff
@@ -47,8 +47,6 @@
 /**
  * A table that can be modified.
  *
- * NOTE: The current API is inefficient and experimental. It will change without notice.
```
[the remark this answers is not in the document]
Comment (vogti) [reply core/src/main/java/org/polypheny/db/schema/ModifiableTable.java]: Yes

File: dbms/src/main/java/org/polypheny/db/schema/PolySchemaBuilder.java [hunk 2]
```diff
@@ -69,19 +83,28 @@ public AbstractPolyphenyDbSchema update( Transaction transaction ) {
         }
 
         for ( CatalogCombinedSchema combinedSchema : combinedDatabase.getSchemas() ) {
-            Map<String, Table> tableMap = new HashMap<>();
+            Map<String, LogicalTable> tableMap = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (vogti) [reply dbms/src/main/java/org/polypheny/db/schema/PolySchemaBuilder.java]: I create a logical schema below. LogicalSchemas are restricted to LogicalTables.

File: jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/JdbcRules.java [hunk 3]
```diff
@@ -168,7 +169,7 @@ public JdbcJoinRule( JdbcConvention out, RelBuilderFactory relBuilderFactory ) {
                     Convention.NONE,
                     out,
                     relBuilderFactory,
-                    "JdbcJoinRule" );
+                    "JdbcJoinRule." + out );
```
[the remark this answers is not in the document]
Comment (vogti) [reply jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/JdbcRules.java]: Yes. Without that, the rule is only registered once but we need it to be registered for each convention.

File: jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/JdbcSchema.java [hunk 4]
```diff
@@ -265,7 +256,7 @@ public Expression getExpression( SchemaPlus parentSchema, String name ) {
 
 
     @Override
-    public Table getTable( String name ) {
+    public JdbcTable getTable( String name ) {
```
[the remark this answers is not in the document]
Comment (vogti) [reply jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/JdbcSchema.java]: I needed that for a something I later managed to solve differently. I reverted it. Nevertheless, from my understanding there would be no issue with returning a JDBC Table here.

File: dbms/src/main/java/org/polypheny/db/schema/PolySchemaBuilder.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (schoenja) [reply dbms/src/main/java/org/polypheny/db/schema/PolySchemaBuilder.java]: Makes sense.

File: jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/JdbcSchema.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (schoenja) [reply jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/JdbcSchema.java]: I am fine with either way, so we can leave it.

**Pull request merged**